addons/mail/static/src/components/notification_list/tests/notification_list_notification_group_tests.js

```diff
@@ -115,37 +115,13 @@ QUnit.test('mark as read', async function (assert) {
         notification_type: 'email', // expected failure type for email message
     });
     const bus = new Bus();
-    bus.on('do-action', null, payload => {
-        assert.step('do_action');
-        assert.strictEqual(
-            payload.action,
-            'mail.mail_resend_cancel_action',
-            "action should be the one to cancel email"
-        );
-        assert.strictEqual(
-            payload.options.additional_context.default_model,
-            'mail.channel',
-            "action should have the group model as default_model"
-        );
-        assert.strictEqual(
-            payload.options.additional_context.unread_counter,
-            1,
-            "action should have the group notification length as unread_counter"
-        );
-    });
     const { createNotificationListComponent } = await this.start({ env: { bus } });
     await createNotificationListComponent();
     assert.containsOnce(
         document.body,
         '.o_NotificationGroup_markAsRead',
         "should have 1 mark as read button"
     );
-
-    document.querySelector('.o_NotificationGroup_markAsRead').click();
-    assert.verifySteps(
-        ['do_action'],
-        "should do an action to display the cancel email dialog"
-    );
```

Comment:
The click needs to be kept, otherwise the test is completely useless. There should be a new assert instead, that checks that notifications are indeed gone after the click.

You'll need to adapt the mock server to reflect the changes done in python for this test to work.

Reply:
Thanks, yes, you're right it is better but it is totally new for me. I now test that the button call the server with the right payload.

Note that the scenario is not fully tested as the server send notification back through the bus to update the client.

Reply:
You need to use mock server to bus simulate notification being sent back.

Reply:
You can check existing methods in that file to get examples, of feel free to ask me on Discord if you need more help.

Reply:
I finally managed to do it. Thanks for your explanation.